packages/common/lib/src/events/default_event_processor.dart

```diff
@@ -50,7 +50,7 @@ final class DefaultEventProcessor implements EventProcessor {
 
   DefaultEventProcessor(
       {required LDLogger logger,
-      bool indexEvents = false,
+      required bool indexEvents,
```

Comment:
For reviewers: I was trying to find a way to not have to tweak this, but the typedef of EventProcessorFactory was not handling the optional named parameter with default well.

Reply:
I am not sure I see why the type here needs to change, it seems like it would just get redundantly passed, which should be fine. Like required at the factory level, but not at the constructor level.

That said I am indifferent to the change.